stock_scheduler_assignation_horizon/models/procurement_group.py

```diff
@@ -2,39 +2,48 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from datetime import datetime, timedelta
-
-import pytz
+from logging import getLogger
 
 from odoo import models
 from odoo.osv import expression
+from odoo.tools.misc import str2bool
+
+_logger = getLogger(__name__)
 
 
 class ProcurementGroup(models.Model):
     _inherit = "procurement.group"
 
     def _get_moves_to_assign_domain(self, company_id):
         domain = super()._get_moves_to_assign_domain(company_id)
-        company = self.env["res.company"].browse(company_id)
-        if company.is_moves_assignation_limited:
-            domain = self._get_moves_to_assign_domain_assignation_limited(
-                domain, company
-            )
+        global_limit = self._get_global_is_moves_assignation_limited()
+        if global_limit:
+            delay = self._get_assignation_horizon_limit()
+            horizon_domain = self._get_assignation_horizon_domain(delay)
+            domain = expression.AND([domain, horizon_domain])
         return domain
 
-    def _get_moves_to_assign_domain_assignation_limited(self, domain, company):
-        company_tz = pytz.timezone(company.partner_id.tz or "UTC")
-        max_date = (
-            datetime.combine(
-                datetime.now(company_tz)
-                + timedelta(days=company.moves_assignation_horizon),
-                datetime.max.time(),
-            )
-            .astimezone(pytz.utc)
-            .replace(tzinfo=None)
-        )
-        return expression.AND(
-            [
-                domain,
-                [("date", "<=", max_date)],
-            ]
+    def _get_assignation_horizon_domain(self, delay_days):
+        max_date = datetime.now() + timedelta(days=delay_days)
+        return [("date", "<=", max_date)]
+
+    def _get_global_is_moves_assignation_limited(self):
```

Comment:
```suggestion
    def _is_assignation_horizon_limited(self):
```
As asked in previous comment :wink: